src/CMakeLists.txt

```
@@ -1007,6 +1007,7 @@ if (MSVC)
         /wd4505
         /wd4100
         /wd4267
+        /Zc:__cplusplus
```

Nit: add comment as to why we're setting this.

```suggestion
        # Enable updated '__cplusplus' macro - workaround for CMake#18837
        /Zc:__cplusplus
```